verifier/src/report_body.rs

```diff
@@ -65,35 +69,43 @@ report_body_field_accessor! {
     ReportData, report_data;
 }
 
-trait IntoVerificationError {
-    fn into_verification_error(expected: Self, actual: Self) -> VerificationError;
+trait IntoVerificationValues<T: VerificationMessage> {
+    fn into_verification_values(expected: Self, actual: Self) -> T;
 }
 
 /// Common implementation for [`Verifier`]s that test for equality between
 /// an expected and actual value.
+///
+/// The `T` is the type to verify.
+/// The `O` is the output data type to use for [`Verifier::VerificationValues`]
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
-pub struct EqualityVerifier<T> {
+pub struct EqualityVerifier<T, O> {
     expected: T,
+    _phantom: core::marker::PhantomData<O>,
```

Comment:
This was my attempt at allowing the `Verifier::VerificationValues` to be generic in the `impl Verifier for EqualityVerifier`

Up for suggestions if there is a better way. My understanding is the `O` has to be bound to either the `Verifier` or the `EqualityVerifier` in the impl.